python/feldera/sql_context.py

```diff
@@ -269,34 +268,33 @@ def register_table_from_sql(self, ddl: str):
 
         self.tables[name] = SQLTable(name, ddl)
 
-    def connect_source_pandas(self, table_name: str, df: pandas.DataFrame):
+    def connect_source_pandas(self, table_name: str, df: pandas.DataFrame, flush: bool = False):
         """
         Adds a pandas DataFrame to the input buffer of the SQLContext, to be pushed to the pipeline.
+        Note that if the pipeline is running, the data will not be pushed if `flush` is False.
 
         :param table_name: The name of the table.
         :param df: The pandas DataFrame to be pushed to the pipeline.
+        :param flush: If True, the data will be pushed to the pipeline immediately. Defaults to False.
         """
 
+        if flush and self.pipeline_status() != PipelineStatus.RUNNING:
```

Comment:
We may want to allow pushing data in the PAUSED state with the `force` flag. This could be a good debugging tool: pause the pipeline, push a small change manually, see what happens. 

You could add an optional `force` argument to this function for that. Ok to make that a separate future PR.